### PR TITLE
Create Welcome react.js

### PR DESCRIPTION
**Pull request by eriemer1**
New parser to respond in a thread when the :welcome: react is used to greet new users and give some general tips on being a good member of the SNDevs slack

### CODE REVIEW

**Comment by eriemer1** [comment]
I am pretty sure I came up with a good way to make it only trigger once per thread, which should solve the concerns in #88

File: Parsers/Welcome react.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+			}
+		}
+	]
+};
+
+//This should find any other payloads where there is already a :welcome: reaction in the thread to prevent the message from firing more than once
+var previousMessage = new global.GlideQuery('x_snc_slackerbot_payload')
+	.where('payload', 'CONTAINS', '"type": "reaction_added"')
+	.where('payload', 'CONTAINS', '"reaction": "welcome"')
+	.where('payload', 'CONTAINS', '"is_bot": true')
```
Comment (chelming) [inline Parsers/Welcome react.js:92]: wouldn't `.where('payload', 'CONTAINS', '"is_bot": true')` filter out reactji's added by people who are totally humans? I think we want this removed so it fires regardless of who added the reatji, right?

File: Parsers/Welcome react.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+/*
+emoji:welcome
+*/
+
+var message = {
+	"type": "home",
```
Comment (chelming) [inline Parsers/Welcome react.js:6]: this looks like it'll throw an error: [block kit builder link](https://app.slack.com/block-kit-builder/T0E2AD6AY#%7B%22type%22:%22home%22,%22blocks%22:%5B%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Hello%20and%20welcome%20to%20SNDevs%20slack!%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Here%20are%20some%20tips%20to%20get%20you%20started*%22%7D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Asking%20Good%20questions%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22-%20Use%20Threads!%20#general%20gets%20busy%20so%20make%20sure%20to%20keep%20conversation%20about%20your%20question%20in%20the%20original%20thread.%20%5Cn-%20Write%20a%20title%20that%20summarizes%20the%20specific%20problem.%20%5Cn-%20Pretend%20you're%20talking%20to%20a%20busy%20colleague.%5Cn%20-Spelling,%20grammar,%20and%20punctuation%20are%20important!%20%5Cn-%20Bad:%20GlideRecord%20isn't%20working%20%5Cn-%20Good:%20Why%20does%20str%20==%20'value'%20evaluate%20to%20false%20when%20str%20is%20set%20to%20'value'%20in%20my%20script?%20%5Cn-%20Proof-read%20before%20posting!%20%5Cn-%20Post%20the%20question%20and%20respond%20to%20feedback%22%7D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Posting%20Code%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22-%20Introduce%20the%20problem%20before%20you%20post%20any%20code.%20%5Cn-%20Help%20others%20reproduce%20the%20problem.%20Don't%20just%20copy%20in%20your%20entire%20script.%20%5Cn-%20Do%20not%20post%20images%20of%20code,%20data,%20error%20messages%20(just%20copy%20the%20text).%5Cn%20-When%20posting%20code%20wrap%20it%20in%201x%20and%203x%20back-tick%20%5B%60%5D%20syntax,%20or%20use%20the%20code%20editor%20%5Cn-%20%60your%20text%60%20=%20Red%20highlighted%20single-line%20%5Cn%20%60%60%60your%20text%60%60%60%20=%20Grey%20highlighted%20multi-line%20%5Cn-%20Or%20use%20the%20%3C/%3E%20code%20button%20on%20the%20text%20editor%20in%20slack.%22%7D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Give%20Points%20to%20say%20thank%20you!%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22If%20someone%20successfully%20helped%20you%20(or%20ever%20just%20help%20you%20get%20closer%20to%20the%20answer)%20tag%20them%20with%20a%20++%20in%20the%20thread.%20eg.%20%3C@'%20+%20current.user.user_id%20+%20'%3E%20++%20will%20grant%20that%20user%20some%20community%20points%20to%20show%20how%20helpful%20they%20were.%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Looking%20for%20more%20helpful%20tips?%20These%20tips%20are%20mostly%20from%20https://stackoverflow.com/help/how-to-ask%22%7D%7D%5D%7D)

Comment (chelming) [reply Parsers/Welcome react.js:6]: nope. nm. this is fine. only message.blocks gets sent.

File: Parsers/Welcome react.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+			}
+		},
+		{
+			"type": "divider"
+		},
+		{
+			"type": "section",
+			"text": {
+				"type": "mrkdwn",
+				"text": "If someone successfully helped you (or ever just help you get closer to the answer) tag them with a ++ in the thread. eg. <@' + current.user.user_id + '> ++ will grant that user some community points to show how helpful they were."
```
Comment (chelming) [inline Parsers/Welcome react.js:72]: the `<@' + current.user.user_id + '>` should be double quotes instead of single quotes

**Comment by chelming** [comment]
I made some small changes to the formatting. Apparently slack's markdown message block won't do bullet points 💀 

Here's a preview of [the following in block-kit builder](https://app.slack.com/block-kit-builder/T0E2AD6AY#%7B%22blocks%22:%5B%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Hello%20and%20welcome%20to%20SNDevs%20slack!%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Here%20are%20some%20tips%20to%20get%20you%20started*%22%7D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Asking%20Good%20questions%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22%20%E2%80%A2%20Use%20Threads!%20#general%20gets%20busy%20so%20make%20sure%20to%20keep%20conversation%20about%20your%20question%20in%20the%20original%20thread.%20%5Cn%20%E2%80%A2%20Write%20a%20title%20that%20summarizes%20the%20specific%20problem.%20%5Cn%20%E2%80%A2%20Pretend%20you're%20talking%20to%20a%20busy%20colleague.%5Cn%20%E2%80%A2%20Spelling,%20grammar,%20and%20punctuation%20are%20important!%20%5Cn%20%E2%80%A2%20Bad:%20GlideRecord%20isn't%20working%20%5Cn%20%E2%80%A2%20Good:%20Why%20does%20str%20==%20'value'%20evaluate%20to%20false%20when%20str%20is%20set%20to%20'value'%20in%20my%20script?%20%5Cn%20%E2%80%A2%20Proof-read%20before%20posting!%20%5Cn%20%E2%80%A2%20Post%20the%20question%20and%20respond%20to%20feedback%22%7D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Posting%20Code%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22%20%E2%80%A2%20Introduce%20the%20problem%20before%20you%20post%20any%20code.%20%5Cn%20%E2%80%A2%20Help%20others%20reproduce%20the%20problem.%20Don't%20just%20copy%20in%20your%20entire%20script.%20%5Cn-%20Do%20not%20post%20images%20of%20code,%20data,%20error%20messages%20(just%20copy%20the%20text).%5Cn%20%E2%80%A2%20When%20posting%20code%20wrap%20it%20in%201x%20and%203x%20back-tick%20%5B%60%5D%20syntax,%20or%20use%20the%20code%20editor%20%5Cn%20%60%60%60%60your%20text%60%20=%20Red%20highlighted%20single-line%60%60%60%20%5Cn%20%60%60%60%60%60%60your%20text%60%60%60%60%60%60%20%60%60%60=%20Grey%20highlighted%20multi-line%60%60%60%20%5Cn%20%E2%80%A2%20Or%20use%20the%20%3C/%3E%20code%20button%20on%20the%20text%20editor%20in%20slack.%22%7D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Give%20Points%20to%20say%20thank%20you!%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22If%20someone%20successfully%20helped%20you%20(or%20ever%20just%20help%20you%20get%20closer%20to%20the%20answer)%20tag%20them%20with%20a%20++%20in%20the%20thread.%20eg.%20%3C@'%20+%20current.user.user_id%20+%20'%3E%20++%20will%20grant%20that%20user%20some%20community%20points%20to%20show%20how%20helpful%20they%20were.%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Looking%20for%20more%20helpful%20tips?%20These%20tips%20are%20mostly%20from%20https://stackoverflow.com/help/how-to-ask%22%7D%7D%5D%7D).

```json
{
	"blocks": [
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "Hello and welcome to SNDevs slack!"
			}
		},
		{
			"type": "divider"
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "*Here are some tips to get you started*"
			}
		},
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "Asking Good questions"
			}
		},
		{
			"type": "divider"
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": " • Use Threads! #general gets busy so make sure to keep conversation about your question in the original thread. \n • Write a title that summarizes the specific problem. \n • Pretend you're talking to a busy colleague.\n • Spelling, grammar, and punctuation are important! \n • Bad: GlideRecord isn't working \n • Good: Why does str == 'value' evaluate to false when str is set to 'value' in my script? \n • Proof-read before posting! \n • Post the question and respond to feedback"
			}
		},
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "Posting Code"
			}
		},
		{
			"type": "divider"
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": " • Introduce the problem before you post any code. \n • Help others reproduce the problem. Don't just copy in your entire script. \n- Do not post images of code, data, error messages (just copy the text).\n • When posting code wrap it in 1x and 3x back-tick [`] syntax, or use the code editor \n ````your text` = Red highlighted single-line``` \n ``````your text`````` ```= Grey highlighted multi-line``` \n • Or use the </> code button on the text editor in slack."
			}
		},
		{
			"type": "header",
			"text": {
				"type": "plain_text",
				"text": "Give Points to say thank you!"
			}
		},
		{
			"type": "divider"
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "If someone successfully helped you (or ever just help you get closer to the answer) tag them with a ++ in the thread. eg. <@" + current.user.user_id + "> ++ will grant that user some community points to show how helpful they were."
			}
		},
		{
			"type": "divider"
		},
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Looking for more helpful tips? These tips are mostly from https://stackoverflow.com/help/how-to-ask"
			}
		}
	]
}
```

File: Parsers/Welcome react.js (hunk 1 above)
Comment (eriemer1) [reply Parsers/Welcome react.js:92]: I checked some payloads and '"is_bot": true' seems to be there for all human posts ¯\_(ツ)_/¯

**Comment by eriemer1** [comment]
Updates made to resolve issues pointed out

**Comment by chelming** [comment]
YOLO!

**Pull request merged**